internal/cephfs/volume.go

```diff
@@ -128,27 +134,23 @@ type localClusterState struct {
 	subVolumeGroupCreated bool
 }
 
-func createVolume(ctx context.Context, volOptions *volumeOptions, cr *util.Credentials, volID volumeID, bytesQuota int64) error {
+func createVolume(ctx context.Context, volOptions *volumeOptions, volID volumeID, bytesQuota int64) error {
 	// verify if corresponding ClusterID key is present in the map,
 	// and if not, initialize with default values(false).
 	if _, keyPresent := clusterAdditionalInfo[volOptions.ClusterID]; !keyPresent {
 		clusterAdditionalInfo[volOptions.ClusterID] = &localClusterState{}
 	}
 
+	ca, err := volOptions.conn.GetFSAdmin()
+	if err != nil {
+		util.ErrorLog(ctx, "could not get FSAdmin, can not create volume %s: %s", volOptions.FsName, err)
+		return err
+	}
+
 	// create subvolumegroup if not already created for the cluster.
 	if !clusterAdditionalInfo[volOptions.ClusterID].subVolumeGroupCreated {
-		err := execCommandErr(
-			ctx,
-			"ceph",
-			"fs",
-			"subvolumegroup",
-			"create",
-			volOptions.FsName,
-			volOptions.SubvolumeGroup,
-			"-m", volOptions.Monitors,
-			"-c", util.CephConfigPath,
-			"-n", cephEntityClientPrefix+cr.ID,
-			"--keyfile="+cr.KeyFile)
+		opts := fsAdmin.SubVolumeGroupOptions{}
```

Comment:
I am wondering , is it the right choice to keep it in fsadmin.

Reply:
what do you mean?

Reply:
discard.